diff --git a/vvp/vpi_modules.cc b/vvp/vpi_modules.cc
index 24fa2a30..9b107a99 100644
--- a/vvp/vpi_modules.cc
+++ b/vvp/vpi_modules.cc
@@ -40,6 +40,33 @@ static unsigned vpip_module_path_cnt = 0;
 
 static bool disable_default_paths = false;
 
+#ifdef __MINGW32__
+static inline char *strndup(const char *str,size_t max)
+{
+      size_t len=max;
+      char *nstr,*zero;
+
+      if (str == (char*)0)
+            return 0;
+
+      zero=(char*)memchr(str, 0, max);
+      if(zero)
+            len=zero-str;
+
+      if(len == (size_t) -1)
+            return (char *)0;
+
+      nstr = (char*)malloc(len + 1);
+      if(!nstr)
+            return (char *)0;
+  
+      memcpy(nstr, str, len);
+      nstr[len] = 0;
+
+      return nstr;
+}
+#endif
+
 void vpip_clear_module_paths()
 {
       vpip_module_path_cnt = 0;
@@ -102,13 +129,13 @@ void vpip_add_env_and_default_module_paths()
       s = strrchr(basepath, '\\');
       if (s) *s = 0;
       else {
-	    fprintf(stderr, "%s: Missing first \\ in exe path!\n", argv[0]);
+	    fprintf(stderr, "Missing first \\ in exe path!\n");
 	    exit(1);
       }
       s = strrchr(basepath, '\\');
       if (s) *s = 0;
       else {
-	    fprintf(stderr, "%s: Missing second \\ in exe path!\n", argv[0]);
+	    fprintf(stderr, "Missing second \\ in exe path!\n");
 	    exit(1);
       }
       strcat(s, "\\lib\\ivl" IVL_SUFFIX);
